[Bug] VcNarrator doesn't work / crashes while starting

### Discord Account

Komkry#0630

### What happens when the bug or crash occurs?

Plugin VcNarrator doesnt work or wont even start

![obrazek](https://github.com/Vendicated/Vencord/assets/78811061/6774d88e-a30e-426a-bcc9-cbe23891c7d1)
![obrazek](https://github.com/Vendicated/Vencord/assets/78811061/d42ca5ae-f28d-43fa-9604-e98c2ea1979b)
![obrazek](https://github.com/Vendicated/Vencord/assets/78811061/8f489e4d-4cdd-4173-ac0e-8fd7acc6d874)

**Vencord**: v1.8.4 • [f74da73](<https://github.com/Vendicated/Vencord/commit/f74da73>) - 15 May 2024
**Client**: stable ~ Discord Desktop v1.0.9147
**Platform**: Win32
**Last Crash Reason**: N/A

Enabled Plugins (22):
```
AlwaysAnimate, BetterSessions, BiggerStreamPreview, CallTimer, ClearURLs, CrashHandler, FakeNitro, ForceOwnerCrown, FriendInvites, FriendsSince, GameActivityToggle, MemberCount, NewGuildSettings, NoDevtoolsWarning, NoSystemBadge, OpenInApp, ReadAllNotificationsButton, RelationshipNotifier, ShowMeYourName, SilentTyping, UserVoiceShow, VolumeBooster
```


### What is the expected behaviour?

Be able to start the plugin and it should work as expected

### How do you recreate this bug or crash?

(after restarting discord)
Go to plugin tab
Search for VcNarraror
Enable (enables)
Try if it does work (doesn't)
Go back to plugin tab
Disable VcNarrator
Enable VcNarrator -> Error while starting plugin VcNarrator

### Errors

```
  Vencord   PluginManager  Starting plugins (stage Init)
 [FAST CONNECT] wss://gateway.discord.gg/?encoding=etf&v=9&compress=zstd-stream, encoding: etf, version: 9
Logger.ts:35  Vencord   WebpackInterceptor  Patching webpackChunkdiscord_app.push
Logger.ts:35  Vencord   WebpackInterceptor  Found Webpack module factory web.d945426e884c33220afa.js
Logger.ts:35  Vencord   WebpackInterceptor  Found main WebpackRequire.onChunksLoaded
Logger.ts:35  Vencord   WebpackInterceptor  Patched initialize app callback invoked, initializing our internal references to WebpackRequire and running beforeInitListeners
VM255 C:\Users\Komkr\AppData\Local\Discord\app-1.0.9147\modules\discord_desktop_core-1\discord_desktop_core\core.asar\app\node_modules\@sentry\utils\cjs\instrument.js:124 Initializing voice engine with audio subsystem: false
42261.88ef0c51b500149123e4.js:1 [default] [BUILD INFO] Release Channel: stable, Build Number: 293362, Version Hash: 73826fe7127010e66ea4c1a0eaad142e989266dc
42261.88ef0c51b500149123e4.js:1 Window state not initialized undefined
I @ 42261.88ef0c51b500149123e4.js:1
42261.88ef0c51b500149123e4.js:1 [default] [NATIVE INFO] host 1.0.9147, modules: discord_media: 1, discord_desktop_core: 1, discord_game_utils: 1, discord_erlpack: 1, discord_spellcheck: 1, discord_cloudsync: 1, discord_zstd: 1, discord_overlay2: 1, discord_modules: 1, discord_krisp: 1, discord_dispatch: 1, discord_voice: 1, discord_rpc: 1, discord_utils: 1, build: 47834
Logger.ts:35  Vencord   WebpackInterceptor  Found Webpack module factory sentry.765b00e66783ff42fca1.js
Logger.ts:35  Vencord   PluginManager  Starting plugins (stage DOMContentLoaded)
VM255 C:\Users\Komkr\AppData\Local\Discord\app-1.0.9147\modules\discord_desktop_core-1\discord_desktop_core\core.asar\app\node_modules\@sentry\utils\cjs\instrument.js:124 Initializing krisp module
42261.88ef0c51b500149123e4.js:1 [Spellchecker] sh is not a valid locale.
(anonymous) @ 42261.88ef0c51b500149123e4.js:1
42261.88ef0c51b500149123e4.js:1 [Spellchecker] Switching to en-US (available)
42261.88ef0c51b500149123e4.js:1 Uncaught (in promise) Error: Cannot find module './notosans-400-normalitalic.woff2'
    at 42261.88ef0c51b500149123e4.js:1:12247
    at async Promise.all (index 3)
 [FAST CONNECT] connected in 2600ms
32612.399fea416275f1e98733.js:12 [DEPRECATED] Please use `subscribeWithSelector` middleware
a @ 32612.399fea416275f1e98733.js:12
42261.88ef0c51b500149123e4.js:1 [RPCServer:WSS] Starting on 6463
42261.88ef0c51b500149123e4.js:1 [GatewaySocket] [CONNECT] wss://gateway.discord.gg, encoding: etf, version: 9, compression: zstd-stream
42261.88ef0c51b500149123e4.js:1 [GatewaySocket] [FAST CONNECT] successfully took over websocket, state: Object
42261.88ef0c51b500149123e4.js:1 [GatewaySocket] [CONNECTED] wss://gateway.discord.gg/?encoding=etf&v=9&compress=zstd-stream in 0 ms
42261.88ef0c51b500149123e4.js:1 [RPCServer:IPC] Starting on \\?\pipe\discord-ipc-0
base.css:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()
42261.88ef0c51b500149123e4.js:1 [GatewaySocket] [READY] took 1954ms, as 527d64affb69b71802837bbcd6792541
42261.88ef0c51b500149123e4.js:1 [Flux] Dispatching CONNECTION_OPEN
42261.88ef0c51b500149123e4.js:1 [DatabaseManager] removing database (user: 248715603215384578, database: undefined)
42261.88ef0c51b500149123e4.js:1 [NativeDispatchUtils] Tried getting Dispatch instance before instantiated
(anonymous) @ 42261.88ef0c51b500149123e4.js:1
42261.88ef0c51b500149123e4.js:1 [Spotify] WS Connecting
42261.88ef0c51b500149123e4.js:1 [Spotify] Added account: komkry
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetching messages for 1026515880080842772 between undefined and undefined. jump={"jumpType":"ANIMATED"}
42261.88ef0c51b500149123e4.js:1 [GatewaySocket] Dispatched 1 messages in 222ms
Logger.ts:35  Vencord   PluginManager  Starting plugins (stage WebpackReady)
Logger.ts:35  Vencord   PluginManager  Starting plugin BadgeAPI
Logger.ts:35  Vencord   PluginManager  Starting plugin BetterSessions
Logger.ts:35  Vencord   PluginManager  Starting plugin ClearURLs
Logger.ts:35  Vencord   PluginManager  Starting plugin FakeNitro
Logger.ts:35  Vencord   PluginManager  Starting plugin GameActivityToggle
Logger.ts:35  Vencord   PluginManager  Starting plugin ReadAllNotificationsButton
Logger.ts:35  Vencord   PluginManager  Starting plugin RelationshipNotifier
Logger.ts:35  Vencord   PluginManager  Starting plugin SilentTyping
42261.88ef0c51b500149123e4.js:1 [GatewaySocket] [READY_SUPPLEMENTAL] took 2216ms
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetched 50 messages for 1026515880080842772 isBefore:false isAfter:false
42261.88ef0c51b500149123e4.js:1 [ChannelMessages] loadComplete: resetting state for channelId=1026515880080842772, sending.length=0
42261.88ef0c51b500149123e4.js:1 [Spotify] WS Connected
42261.88ef0c51b500149123e4.js:1 [Spotify] Devices updated for komkry: Array(0)
42261.88ef0c51b500149123e4.js:1 [Spotify] Profile updated for komkry: isPremium = true
Logger.ts:35  Vencord   PluginManager  Starting plugin VcNarrator
Logger.ts:35  Vencord   PluginManager  Starting plugin VcNarrator
Logger.ts:35  Vencord   PluginManager  VcNarrator already started
_log @ Logger.ts:35
Logger.ts:35  Vencord   PluginSettings  Error while starting plugin VcNarrator
_log @ Logger.ts:35
Logger.ts:35  Vencord   PluginManager  Starting plugin VcNarrator
Logger.ts:35  Vencord   PluginManager  VcNarrator already started
_log @ Logger.ts:35
Logger.ts:35  Vencord   PluginSettings  Error while starting plugin VcNarrator
_log @ Logger.ts:35
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetching messages for 801937699195584553 between undefined and undefined. jump={"jumpType":"ANIMATED"}
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetched 3 messages for 801937699195584553 isBefore:false isAfter:false
42261.88ef0c51b500149123e4.js:1 [ChannelMessages] loadComplete: resetting state for channelId=801937699195584553, sending.length=0
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetching messages for 1116439202834423939 between undefined and undefined. jump={"jumpType":"ANIMATED"}
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetching messages for 1039585467584229479 between undefined and undefined. jump={"jumpType":"ANIMATED"}
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetched 50 messages for 1116439202834423939 isBefore:false isAfter:false
42261.88ef0c51b500149123e4.js:1 [ChannelMessages] loadComplete: resetting state for channelId=1116439202834423939, sending.length=0
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetched 50 messages for 1039585467584229479 isBefore:false isAfter:false
42261.88ef0c51b500149123e4.js:1 [ChannelMessages] loadComplete: resetting state for channelId=1039585467584229479, sending.length=0
42261.88ef0c51b500149123e4.js:1 [Routing/Utils] transitionTo - Transitioning to /channels/1015060230222131221/1026515880080842772
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetching messages for 1026515880080842772 between 1240280701107044454 and undefined. jump=undefined
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetched 50 messages for 1026515880080842772 isBefore:true isAfter:false
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Queueing message to be sent LogId:177
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Draining message from queue LogId:177 QueueLength: 0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Draining Message Queue with:  0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Finished draining message from queue LogId:177 QueueLength: 0
42261.88ef0c51b500149123e4.js:1 [ComponentDispatchUtils] ComponentDispatch.subscribe: Attempting to add a duplicate listener POPOUT_CLOSE
(anonymous) @ 42261.88ef0c51b500149123e4.js:1
/api/v9/users/@me/notes/921310260189335603:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()
42261.88ef0c51b500149123e4.js:1 [Routing/Utils] transitionTo - Transitioning to /channels/1015060230222131221/1039585467584229479
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Queueing message to be sent LogId:1983
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Draining message from queue LogId:1983 QueueLength: 0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Draining Message Queue with:  0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Finished draining message from queue LogId:1983 QueueLength: 0
42261.88ef0c51b500149123e4.js:1 [ComponentDispatchUtils] ComponentDispatch.subscribe: Attempting to add a duplicate listener POPOUT_CLOSE
(anonymous) @ 42261.88ef0c51b500149123e4.js:1
/api/v9/users/@me/notes/797721161504653332:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()
cdn.discordapp.com/avatars/976176454511509554/facb5fc93e32477dcb99bb618b11d02b.webp?size=20:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetching messages for 942209091462365214 between undefined and undefined. jump={"jumpType":"ANIMATED"}
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetched 50 messages for 942209091462365214 isBefore:false isAfter:false
42261.88ef0c51b500149123e4.js:1 [ChannelMessages] loadComplete: resetting state for channelId=942209091462365214, sending.length=0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Queueing message to be sent LogId:438
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Draining message from queue LogId:438 QueueLength: 0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Draining Message Queue with:  0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Finished draining message from queue LogId:438 QueueLength: 0
42261.88ef0c51b500149123e4.js:1 [ComponentDispatchUtils] ComponentDispatch.subscribe: Attempting to add a duplicate listener POPOUT_CLOSE
(anonymous) @ 42261.88ef0c51b500149123e4.js:1
/api/v9/users/@me/notes/712653921692155965:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Queueing message to be sent LogId:8643
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Draining message from queue LogId:8643 QueueLength: 0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Draining Message Queue with:  0
42261.88ef0c51b500149123e4.js:1 [MessageQueue] Finished draining message from queue LogId:8643 QueueLength: 0
42261.88ef0c51b500149123e4.js:1 [Routing/Utils] transitionTo - Transitioning to /channels/1015060230222131221/1222936386626129920
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetching messages for 1222936386626129920 between undefined and undefined. jump={"jumpType":"ANIMATED"}
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetched 2 messages for 1222936386626129920 isBefore:false isAfter:false
42261.88ef0c51b500149123e4.js:1 [ChannelMessages] loadComplete: resetting state for channelId=1222936386626129920, sending.length=0
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetching messages for 194905128296382464 between undefined and undefined. jump={"jumpType":"ANIMATED"}
42261.88ef0c51b500149123e4.js:1 [discord_protos.discord_users.v1.FrecencyUserSettings] Loading proto
42261.88ef0c51b500149123e4.js:1 [MessageActionCreators] Fetched 13 messages for 194905128296382464 isBefore:false isAfter:false
42261.88ef0c51b500149123e4.js:1 [ChannelMessages] loadComplete: resetting state for channelId=194905128296382464, sending.length=0
base.css:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()
Logger.ts:35  Vencord   PluginManager  Starting plugin VcNarrator
Logger.ts:35  Vencord   PluginManager  VcNarrator already started
_log @ Logger.ts:35
warn @ Logger.ts:61
(anonymous) @ index.ts:163
ll.<computed> @ lazy.ts:56
d @ index.tsx:138
onChange @ Switch.tsx:72
eB @ 32612.399fea416275f1e98733.js:12
eG @ 32612.399fea416275f1e98733.js:12
(anonymous) @ 32612.399fea416275f1e98733.js:12
re @ 32612.399fea416275f1e98733.js:12
rt @ 32612.399fea416275f1e98733.js:12
(anonymous) @ 32612.399fea416275f1e98733.js:12
sA @ 32612.399fea416275f1e98733.js:12
eI @ 32612.399fea416275f1e98733.js:12
rs @ 32612.399fea416275f1e98733.js:12
tB @ 32612.399fea416275f1e98733.js:12
tP @ 32612.399fea416275f1e98733.js:12
Logger.ts:35  Vencord   PluginSettings  Error while starting plugin VcNarrator
_log @ Logger.ts:35
error @ Logger.ts:53
d @ index.tsx:144
onChange @ Switch.tsx:72
eB @ 32612.399fea416275f1e98733.js:12
eG @ 32612.399fea416275f1e98733.js:12
(anonymous) @ 32612.399fea416275f1e98733.js:12
re @ 32612.399fea416275f1e98733.js:12
rt @ 32612.399fea416275f1e98733.js:12
(anonymous) @ 32612.399fea416275f1e98733.js:12
sA @ 32612.399fea416275f1e98733.js:12
eI @ 32612.399fea416275f1e98733.js:12
rs @ 32612.399fea416275f1e98733.js:12
tB @ 32612.399fea416275f1e98733.js:12
tP @ 32612.399fea416275f1e98733.js:12
```

### Request Agreement

- [X] I am using Discord Stable or tried on Stable and this bug happens there as well
- [X] I have read the requirements for opening an issue above

--- a/src/plugins/index.ts
+++ b/src/plugins/index.ts
@@ -170,7 +170,9 @@ export const startPlugin = traceFunction("startPlugin", function startPlugin(p: 
             return false;
         }
     }
+
     p.started = true;
+
     if (commands?.length) {
         logger.debug("Registering commands of plugin", name);
         for (const cmd of commands) {
@@ -200,6 +202,7 @@ export const startPlugin = traceFunction("startPlugin", function startPlugin(p: 
 
 export const stopPlugin = traceFunction("stopPlugin", function stopPlugin(p: Plugin) {
     const { name, commands, flux, contextMenus } = p;
+
     if (p.stop) {
         logger.info("Stopping plugin", name);
         if (!p.started) {
@@ -213,6 +216,7 @@ export const stopPlugin = traceFunction("stopPlugin", function stopPlugin(p: Plu
             return false;
         }
     }
+
     p.started = false;
 
     if (commands?.length) {

--- a/src/plugins/index.ts
+++ b/src/plugins/index.ts
@@ -165,13 +165,12 @@ export const startPlugin = traceFunction("startPlugin", function startPlugin(p: 
         }
         try {
             p.start();
-            p.started = true;
         } catch (e) {
             logger.error(`Failed to start ${name}\n`, e);
             return false;
         }
     }
-
+    p.started = true;
     if (commands?.length) {
         logger.debug("Registering commands of plugin", name);
         for (const cmd of commands) {
@@ -209,12 +208,12 @@ export const stopPlugin = traceFunction("stopPlugin", function stopPlugin(p: Plu
         }
         try {
             p.stop();
-            p.started = false;
         } catch (e) {
             logger.error(`Failed to stop ${name}\n`, e);
             return false;
         }
     }
+    p.started = false;
 
     if (commands?.length) {
         logger.debug("Unregistering commands of plugin", name);

